homeassistant/components/motionblinds_ble/__init__.py

```diff
@@ -0,0 +1,102 @@
+"""Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from functools import partial
+import logging
+
+from motionblindsble.const import MotionBlindType
+from motionblindsble.crypt import MotionCrypt
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.bluetooth import (
+    BluetoothCallbackMatcher,
+    BluetoothChange,
+    BluetoothScanningMode,
+    BluetoothServiceInfoBleak,
+    async_ble_device_from_address,
+    async_register_callback,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [
+    Platform.COVER,
+]
+
+CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Motionblinds BLE integration."""
+
+    _LOGGER.debug("Setting up Motionblinds BLE integration")
+
+    # The correct time is needed for encryption
+    _LOGGER.debug("Setting timezone for encryption: %s", hass.config.time_zone)
+    MotionCrypt.set_timezone(hass.config.time_zone)
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Motionblinds BLE device from a config entry."""
+
+    _LOGGER.debug("(%s) Setting up device", entry.data[CONF_MAC_CODE])
+
+    ble_device = async_ble_device_from_address(hass, entry.data[CONF_ADDRESS])
+    device = MotionDevice(
+        ble_device if ble_device is not None else entry.data[CONF_ADDRESS],
+        blind_type=MotionBlindType[entry.data[CONF_BLIND_TYPE].upper()],
+    )
+
+    # Register Home Assistant functions to use in the library
+    device.set_ha_create_task(
```

Comment:
I would rename this to set_create_task_factory since the library shouldn't know about HA